### DISCUSSION TITLE
Hardhat plugins update: boosting security and developer experience!

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#34] [Announcements]
We're thrilled to announce the release of a range of updates in our [zkSync Era plugins for Hardhat](https://era.zksync.io/docs/tools/hardhat/plugins.html). These updates bring a host of improvements, focusing on enhancing security and streamlining your development process. Here's what you need to know:

- **hardhat-zksync-solc v0.4.0**:
  - Compiler version check: Our plugin now performs version checks, encouraging you to use the latest zksolc versions and ensuring you use the minimum required version available.
  - Zero-config: Simplify your setup! Just import the package in your hardhat.config.ts file, and it will automatically leverage the latest zksolc version. Advanced configurations, such as defining libraries or optimizer parameters, work seamlessly, just as in previous versions.
  - Support for `version: "latest": Enjoy the convenience of easily specifying the latest version of zksolc.
  - The latest version of the zksolc compiler now supports Solidity 0.8.20.

- **hardhat-zksync-vyper v0.2.0**:
  - Compiler version check: Similar to hardhat-zksync-solc, we now offer compiler version checks for seamless integration with the latest zkvyper compiler.
  - The latest version of the zkvyper compiler now supports Vyper 0.3.9.

- **hardhat-zksync-verify v0.1.8**:
  - Support for encoded constructor params for verification.
  - Support for proxy contracts verification.

- **hardhat-zksync-upgradable v0.1.0-beta.2**:
  - First `beta` release of this plugin. 
  - Support for storage validations.
  - Support for proxy contracts verification.

We've also updated our documentation to provide comprehensive details on these changes.

Explore the enhanced security and developer experience that our zkSync Era plugins for Hardhat offer. Happy coding!

**Comment by Kivamar** [comment]
This is very important update! thank you

**Comment by keytwojohn** [comment]
Great to see the release of these updates for zkSync Era plugins for Hardhat! These improvements signify a significant step forward in enhancing security and streamlining the development process. The addition of compiler version checks ensures compatibility with the latest zksolc and zkvyper compilers, promoting best practices and reducing potential compatibility issues. The zero-config feature simplifies setup and allows developers to seamlessly leverage the latest compiler versions without manual intervention. Moreover, the added support for specifying the latest version of zksolc and zkvyper further enhances convenience. The inclusion of support for Solidity 0.8.20 and Vyper 0.3.9 ensures compatibility with the latest language features and optimizations. Additionally, the new features introduced in hardhat-zksync-verify and hardhat-zksync-upgradable plugins, such as support for encoded constructor params, proxy contracts verification, and storage validations, expand the capabilities of zkSync Era development. The updated documentation will undoubtedly aid developers in understanding and utilizing these enhancements effectively. Overall, these updates represent a significant advancement in providing a more secure and efficient development environment for zkSync Era projects. Exciting times ahead for zkSync Era developers!

**Comment by steyn52** [comment]
Great news! These updates to the zkSync Era plugins for Hardhat introduce valuable improvements, including compiler version checks, zero-config setup, support for latest compiler versions, and additional features like encoded constructor params support and proxy contracts verification. These enhancements aim to streamline development processes and enhance security. Developers can now enjoy an improved experience while building on zkSync. Happy coding!

**Comment by cameronstroman** [comment]
Exciting updates in zkSync Era plugins for Hardhat! Improved security and developer experience with version checks, zero-config setup, and support for latest compiler versions. Explore now! Happy coding!

**Comment by deosovitis** [comment]
The zkSync Era plugins for Hardhat have been updated to improve security and streamline development processes. The updates include version checks for zksolc and zkvyper compilers, support for Solidity 0.8.20 and Vyper 0.3.9, and enhancements to various other features. The documentation has also been updated to provide comprehensive details on these changes.
